src/renderers/shaders/ShaderChunk/emissivemap_pars_fragment.glsl

```diff
@@ -2,4 +2,10 @@
 
 	uniform sampler2D emissiveMap;
 
+	vec4 emissiveMapTexelToLinear( vec4 value ) {
+		#define MACRO_DECODE EMISSIVEMAP_ENCODING
+			#include <encoding_template>
```

Comment:
I guess one approach to moving this to JavaScript more would be to just define make "mapTexelDecode" a define which resolves into a macro that calls the decoder function we are looking to use.  I think that could work.  Removes the need for the "include" system though....